internal/node/node.go

```diff
@@ -0,0 +1,75 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Node interface {
+	IsNodeSchedulable(node *v1.Node) bool
+	GetNodesListBySelector(ctx context.Context, mod *kmmv1beta1.Module) ([]v1.Node, error)
```

Comment:
```suggestion
	GetNodesListBySelector(ctx context.Context, selector map[string]string) ([]v1.Node, error)
```

Since this is a generic `Node` package, let's try to make it as less dependent of other custom objects like `module` if possible.
(`module` doesn't exists in the context of k8s node's actions).

Reply:
@ybettan done